docs/Explanation/ADR/adr-004-did.md

```diff
@@ -0,0 +1,687 @@
+            "type": "EcdsaSecp256k1RecoveryMethod2020",
+            "controller": "did:key:cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+            "blockchainAccountID": "cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj"
+        }
+    ],
+    "authentication": [
+        "did:key:cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj"
+    ]
+}
+
+/* DID metadata */
+{
+  "created": "2021-03-23T06:35:22Z",
+  "updated": "2021-03-23T06:35:22Z", 
+  "versionId": "ae11692325525e82337167fcfab34d45d1904ff786e2d4bf4be2d1c4878cd34c" /* hex(blake2b(tx)) */
+}
+
+```
+
+The [`did:key`](https://w3c-ccg.github.io/did-method-key/) method is supported by the module and resolves automatically blockchain addresses. 
+
+It is RECOMMENDED to use an id that is not equals to the blockchain account address for privacy concerns and to isolate the verification methods to the did subject (for example during key rotation)
+
+
+> NOTE: a more fine grained way of creating a DID MAY be implemented with a `MsgCreateDidDocumentWitOptions` with the goal of saving in gas by executing a single transactions in a complex DID scenario.
+
+
+###### Resolve/Verify
+
+The integrity of the DID documents stored on the ledger are guaranteed by the underlying blockchain protocol. 
+
+A DID MAY be resolved using the gRPC message:
+
+```
+QueryDidDocumentRequest(did string)
+```
+
+Example
+```javascript
+/* gRPC message */
+QueryDidDocumentRequest("did:cosmos:cash:806e557e-ecdb-4e80-ab0d-a82ad35c9ceb")
+
+/* DID document */
+{
+    "context": [
+        "https://www.w3.org/ns/did/v1"
+    ],
+    "id": "did:cosmos:cash:806e557e-ecdb-4e80-ab0d-a82ad35c9ceb",
+    "verificationMethods" : [
+        {
+            "id": "did:key:cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj#cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+            "type": "EcdsaSecp256k1RecoveryMethod2020",
+            "controller": "did:key:cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+            "blockchainAccountID": "cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj"
+        }
+    ],
+    "authentication": [
+        "did:key:cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj"
+    ]
+}
+
+/* DID metadata */
+{
+  "created": "2021-03-23T06:35:22Z",
+  "updated": "2021-03-23T06:35:22Z", 
+  "versionId": "ae11692325525e82337167fcfab34d45d1904ff786e2d4bf4be2d1c4878cd34c" /* hex(blake2b(tx)) */
+}
+```
+
+Note that the representation is not compatible with the JSON-LD standard due to some specificity of the Protobuf message format.
+
+###### Update
+
+There are two ways of updating a DID document:
+
+- manage DID controllers
+- manipulate verification methods and relationships 
+
+In both cases the target DID MUST exists on chain and the `signerAccount` MUST exists as a verification method (property `blockchainAccountID`) in a verification relationship of type `authentication` or being listed as a DID controller.
+
+
+**DID controllers** 
```

Comment:
The heading capitalization style is to use Title case, it's handy to use https://capitalizemytitle.com/# 

because we gave two methods above, let's use the same language in the heading